pkg/dcgmexporter/gpu_collector.go

```diff
@@ -283,6 +284,16 @@ func ToMetric(
 
 		gpuModel := getGPUModel(d, replaceBlanksInModelName)
 
+		attrs := map[string]string{}
+		if counter.FieldName == "DCGM_FI_DEV_XID_ERRORS" {
+			attrs["err_code"] = strconv.Itoa(int(val.Int64()))
+			if val.Int64() < 256 && xidErrCodeToText[val.Int64()] != "" {
```

Comment:
Should this value be 143 - the highest reported XID?

Reply:
I was using 256 because I declared `xidErrCodeToText` as an fixed-size array of 256 in size. but maybe I could try using len(xidErrCodeToText) and make `xidErrCodeToText` an slice.

Reply:
Hi @glowkey, I pushed some new changes to covert `xidErrCodeToText` to a slice and has changed the sanity check here accordingly.